client_endpoint.bal

```diff
@@ -0,0 +1,613 @@
+                http:HTTP_GET, requestPath);
+
+        http:Response response = <http:Response> check self.httpClient->get(requestPath, request);
+        [json, ResponseMetadata] jsonResponse = check mapResponseToTuple(response);
+        return mapJsonToDatabaseType(jsonResponse);
+    }
+
+    # List information of all databases in an Azure Cosmos DB account.
+    # 
+    # + maxItemCount - Optional. Maximum number of Databases in the returning stream.
+    # + return - If successful, returns stream<cosmosdb:Database>. else returns error. 
+    remote function listDatabases(int? maxItemCount = ()) returns @tainted stream<Database>|error {
+        http:Request request = new;
+        string requestPath = prepareUrl([RESOURCE_TYPE_DATABASES]);
+
+        check setMandatoryHeaders(request, self.host, self.masterOrResourceToken, self.tokenType, self.tokenVersion, 
+                http:HTTP_GET, requestPath);
+        if (maxItemCount is int) {
+            request.setHeader(MAX_ITEM_COUNT_HEADER, maxItemCount.toString());
+        }
+
+        Database[] emptyArray = [];
+        stream<Database> databaseStream = <stream<Database>> check retriveStream(self.httpClient, requestPath, request, 
+                emptyArray, maxItemCount);
+        return databaseStream;
+    }
+
+    # Retrive information about a container in a database.
+    # 
+    # + databaseId - ID of the database which container belongs to.
+    # + containerId - ID of the container to retrive infromation.  
+    # + requestOptions - Optional. The ResourceReadOptions which can be used to add addtional capabilities to the request.
+    # + return - If successful, returns cosmosdb:Container. Else returns error.  
+    remote function getContainer(string databaseId, string containerId, ResourceReadOptions? requestOptions = ()) 
+            returns @tainted Container|error {
+        http:Request request = new;
+        check createRequest(request, requestOptions);
+        string requestPath = prepareUrl([RESOURCE_TYPE_DATABASES, databaseId, RESOURCE_TYPE_COLLECTIONS, containerId]);
+        check setMandatoryHeaders(request, self.host, self.masterOrResourceToken, self.tokenType, self.tokenVersion, 
+                http:HTTP_GET, requestPath);
+
+        http:Response response = <http:Response> check self.httpClient->get(requestPath, request);
+        [json, ResponseMetadata] jsonResponse = check mapResponseToTuple(response);
+        return mapJsonToContainerType(jsonResponse);
+    }
+
+    # List information of all containers in a database
+    # 
+    # + databaseId - ID of the database where the containers belong to.
+    # + maxItemCount - Optional. Maximum number of Containers to in the returning stream.
+    # + return - If successful, returns stream<cosmosdb:Container>. Else returns error.  
+    remote function listContainers(string databaseId, int? maxItemCount = ()) returns @tainted stream<Container>|error {
```

Comment:
I have included this because there is a pagination related to the returned results. So that we can provide a count of items to be returned in one page. When considering about the streams it is not necessary. I hope to get some idea in the review with ballerina team about this.